helpers/object.ts

```diff
@@ -16,14 +16,26 @@ export default class HelperObject {
         return names;
     }
 
+    /// Turns {a: 1, b: {x: 2, y: 3}} into [['a'], ['b'], ['b','x'], ['b','y']].
+    /// To exclude ['b'], pass includeObjects=false.
     public static recursiveKeys(obj: any, includeObjects = true): string[][] {
         return Object.keys(obj).map(k => {
             let keys: string[][] = [[k]];
             if (typeof(obj[k]) === "object") {
-                const newKeys = this.recursiveKeys(obj[k]).map(_k => [k].concat(_k));
+                const newKeys = this.recursiveKeys(obj[k], includeObjects).map(_k => [k].concat(_k));
                 keys = includeObjects ? keys.concat(newKeys) : newKeys;
             }
             return keys;
         }).reduce((p, v) => p.concat(v), []);
     }
+
+    /// Top-level object can be anything (including a function), but recursively looks only into
+    /// objects (including arrays).
+    public static forEachPath(obj: any, fn: (path: string[], obj: any, key: string) => void, prefix: string[] = []): void {
+      Object.keys(obj).forEach(k => {
```

Comment:
Why not just use pre-existing `recursiveKeys()`? For that matter, why not do something like this:
```typescript
public static recursiveKeyValues(obj: any, includeObjects = true): {
    key: string[];
    value: any;
}[] {
    return this.recursiveKeys(obj, includeObjects).forEach(key => ({
        key,
        value: keys.reduce((p, v) => p[v], obj)
    });
}
```

Then you'd be able to iterate over that return value easily.

Reply:
If you want to change the value so it changes the original object (which was the actual use case here), it is more convenient to have `obj` and `key` the way it is in the original method. I mainly found that the code in the PermissionTransformer was hard to read, and wrote a helper method that is simpler in logic than `recursiveKeys` and leads to clearer code in the PermissionTransformer.

It is less general though and I don't mind it either way. I can just delete this method, if you'd like to.

Reply:
I'm not sure how I feel about it. Maybe this pull request should just include the bugfix and additional test case, and we revisit this another time?